core/src/main/kotlin/behavior/channel/threads/ForumChannelThreadBehavior.kt

```diff
@@ -0,0 +1,28 @@
+package dev.kord.core.behavior.channel.threads
+
+import dev.kord.core.cache.data.toData
+import dev.kord.core.entity.channel.Channel
+import dev.kord.core.entity.channel.thread.ForumChannelThread
+import dev.kord.core.entity.channel.thread.ThreadChannel
+import dev.kord.rest.builder.channel.thread.ForumThreadModifyBuilder
+import dev.kord.rest.builder.channel.thread.ThreadModifyBuilder
+import kotlin.contracts.InvocationKind
+import kotlin.contracts.contract
+
+public interface ForumChannelThreadBehavior : ThreadChannelBehavior
+
+/**
+ * * Editing a thread to set [archived][ThreadChannel.isArchived] to false only requires the current user to be in the thread.
+ * * If [locked][ThreadChannel.isLocked] is true, then the user must have [Manage Threads][dev.kord.common.entity.Permission.ManageThreads]
+ * * Editing a thread to change the
+ * [name][ThreadModifyBuilder.name],
+ * [archived][ThreadModifyBuilder.archived],
+ * [autoArchiveDuration][ThreadModifyBuilder.autoArchiveDuration] fields
+ * requires [Manage Threads][dev.kord.common.entity.Permission.ManageThreads] or that the current user is the thread creator.
+ */
+public suspend inline fun ForumChannelThreadBehavior.edit(builder: ForumThreadModifyBuilder.() -> Unit): ForumChannelThread {
```

Comment:
Overload resolution ambiguity, because `ThreadChannelBehavior.edit` function, I thought of creating an interface called `TextThreadChannelBehavior` and moving `ThreadChannelBehavior.edit` to it, while `ForumChannelThreadBehavior` would inherit `ThreadChannelBehavior`. Is it correct to do this?

Reply:
We should go with a generic builder approach to resolve these nasty sort of errors. But that's beyond this PR's scope

Reply:
ok turns out this ambiguity only exists because the receiver of the passed lambda differs (`ForumThreadModifyBuilder` vs `ThreadModifyBuilder`), if it was the same there would be no ambiguity

should we just have one builder for both (meaning that you could some unsafe things when editing normal threads)?

Reply:
I believe, given hopes goal to change to generic builders, this would be a good solution for now.

Reply:
The solution for this is to move to Generic Builders as I added in Discord, I think it's better to add this to an issue.
But this will mean:
* Less code to maintain
* Builders resuability
* Solves these kind of issues

While we try to enforce types, the user still can make mistakes as we have seen with slash commands
So the whole point of having "Type enforced builders" doesn't stand a chance against the challenges and updates Discord have undergone

Reply:
A temporary solution needs to be found though, you said yourself that this is out of scope for this pr

Reply:
yeah it's ok to add them together.
Just stating that this is a good indicator we need to re-evaluate our abstractions